### PR TITLE
New ember debug command

### PR DESCRIPTION
**Pull request by sivakumar-kailasam**
[RFC details](https://github.com/sivakumar-kailasam/rfcs/blob/master/active/0000-ember-debug-command.md)

### CODE REVIEW

**Comment by bcardarella** [comment]
@sivakumar-kailasam I suspect some of the analytics stuff that @twokul wrote could be purposed for this.

File: active/0000-ember-debug-command.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+- npm version
+- node version
+- OS
+- bower dependencies 
+- npm dependencies
+- Brocfile.js
+- config/environment.js
+- Files counts in node_modues, vendor, bower_components, app folders
+- Output of a command. i.e, sometimes running `ember build` might cause the issue, so when running this command users could possibly pass an argument `--command=`(e.g. --command=debug) which we can run and include the output along with the report.
+
```
Comment (stefanpenner) [inline active/0000-ember-debug-command.md:31]: what format should the output be?
Ideally we optimize for readability as well.

Comment (stefanpenner) [inline active/0000-ember-debug-command.md:31]: we should also detect and log.
- watchman
- path/to/node
- path/to/npm
- node_modules that exist above the current cwd()

**Comment by stefanpenner** [comment]
:+1:

File: active/0000-ember-debug-command.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# Unresolved questions
+
```
Comment (stefanpenner) [inline active/0000-ember-debug-command.md:33]: non i can think of.

**Comment by sivakumar-kailasam** [comment]
@stefanpenner I was thinking of markdown since it would be easy to drag drop and good for readability in the issues. Here's a sample I generated from my ember-cli repo,
https://gist.github.com/sivakumar-kailasam/44b31ea3cf3bde3813d2

If this output looks good I'll raise a PR

**Comment by sivakumar-kailasam** [comment]
@bcardarella  The analytics module being a wrapper over insight might help in looking at stats over the user base anonymously and is currently limited for viewing to folks from the core team.  May be we can start writing these analytics' tracking messages to a file so that whenever `ember debug` is run it could include those messages as well. This PR aims at providing a file in an issue so that the details are available to anyone viewing the issue.

@stefanpenner thoughts on this?

**Comment by twokul** [comment]
hm, I think build time of the trees + errors should be a part of `ember-debug` file. Analytic in Ember CLI s is just for gathering stats, it won't help you much to debug.

**Comment by stefanpenner** [comment]
superseeded by ember doctor

**Pull request closed without merging**